packages/nitro-protocol/test/contracts/ninja-nitro/SingleChannelAdjudicator/claim.test.ts

```diff
@@ -0,0 +1,337 @@
+    ${'14. guarantee not on chain'}                     | ${{g: 5}}  | ${[]}                 | ${{A: 5, B: 5}}       | ${[]}   | ${{A: 5, B: 5}}       | ${{g: 0}} | ${{B: 5}}       | ${incorrectStorage}
+    ${'15. swap guarantee, overfunded, 2 destinations'} | ${{g: 12}} | ${['B', 'A']}         | ${{A: 5, B: 5}}       | ${[]}   | ${{A: 0, B: 0}}       | ${{g: 2}} | ${{A: 5, B: 5}} | ${undefined}
+    ${'16. underspecified guarantee, overfunded      '} | ${{g: 12}} | ${['B']}              | ${{A: 5, B: 5}}       | ${[]}   | ${{A: 5, B: 0}}       | ${{g: 7}} | ${{B: 5}}       | ${undefined}
+  `(
+    '$name',
+    async ({
+      name,
+      heldBefore,
+      guaranteeDestinations,
+      tOutcomeBefore,
+      indices,
+      tOutcomeAfter,
+      heldAfter,
+      payouts,
+      reason,
+    }: {
+      name;
+      heldBefore: AssetOutcomeShortHand;
+      guaranteeDestinations;
+      tOutcomeBefore: AssetOutcomeShortHand;
+      indices: number[];
+      tOutcomeAfter: AssetOutcomeShortHand;
+      heldAfter: AssetOutcomeShortHand;
+      payouts: AssetOutcomeShortHand;
+      reason;
+    }) => {
+      const target = new TestChannel(getRandomNonce(name));
+      const guarantor = new TestChannel(getRandomNonce(name + 'g'));
+      addresses.t = target.id;
+      addresses.g = guarantor.id;
+      // Transform input data (unpack addresses and BigNumber amounts)
+      [heldBefore, tOutcomeBefore, tOutcomeAfter, heldAfter, payouts] = [
+        heldBefore,
+        tOutcomeBefore,
+        tOutcomeAfter,
+        heldAfter,
+        payouts,
+      ].map(object => replaceAddressesAndBigNumberify(object, addresses) as AssetOutcomeShortHand);
+      guaranteeDestinations = guaranteeDestinations.map(x => addresses[x]);
+      // Fund the guarantor channel
+      let gasUsed;
+      gasUsed = await guarantor.depositETH(heldBefore[guarantor.id]);
+      await writeGasConsumption(
+        'SingleChannelAdjudicator.claim.gas.md',
+        name + ': send ETH to Guarantor',
+        gasUsed
+      );
+      // DEPLOY GUARANTOR CHANNEL
+      gasUsed = await guarantor.deploy();
+      await writeGasConsumption(
+        'SingleChannelAdjudicator.claim.gas.md',
+        name + ': deploy Guarantor',
+        gasUsed
+      );
+      // Compute an appropriate guarantee for the guarantor (using only ETH)
+      const guarantee = {
+        destinations: guaranteeDestinations,
+        targetChannelId: target.id,
+      };
+      const guarantorOutcome: Outcome = [{assetHolderAddress: constants.AddressZero, guarantee}];
+      if (guaranteeDestinations.length > 0) {
+        // CONCLUDE GUARANTOR CHANNEL
+        gasUsed = await guarantor.conclude(guarantorOutcome);
+        await writeGasConsumption(
+          'SingleChannelAdjudicator.claim.gas.md',
+          name + ': conclude Guarantor',
+          gasUsed
+        );
+      } else guarantor.outcome = guarantorOutcome; // Set this so that the claim tx should revert in the way we expect
```

Comment:
Nit, and maybe subjective, but causes fewer question marks to spontaneous appear over my head:
```suggestion
      } else {
        // Set this so that the claim tx should revert in the way we expect
        guarantor.outcome = guarantorOutcome;
      }
```